manifest/autoversion/html.go

```diff
@@ -30,14 +31,72 @@ func htmlAutoVersion(client *http.Client, autoVersion *manifest.AutoVersionBlock
 	if err != nil {
 		return "", errors.Wrapf(err, "%s: could not parse HTML", url)
 	}
-	expr, err := xpath.Compile(autoVersion.HTML.XPath)
+
+	var matcher htmlMatcher
+	switch {
+	case autoVersion.HTML.XPath != "":
+		matcher, err = compileXPathMatcher(autoVersion.HTML.XPath)
+		if err != nil {
+			return "", err
+		}
+
+	case autoVersion.HTML.CSS != "":
+		matcher, err = compileCSSMatcher(autoVersion.HTML.CSS)
+		if err != nil {
+			return "", err
+		}
+
+	default:
+		return "", errors.Errorf("must specify either xpath or css for auto-version html")
+	}
+
+	candidates, err := matcher.FindAll(node)
+	if err != nil {
+		return "", err
+	}
+
+	// Parse and sort versions so we can get the latest.
+	versions := make(manifest.Versions, 0, len(candidates))
```

Comment:
I realise the original didn't have this, but would you mind inserting a check that there are >0 candidates. If there aren't, this should error.

Reply:
Makes sense, and good catch! This was probably a panic otherwise.